### PR TITLE
[CELEBORN-850][INFRA] Add SBT CI

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

This PR adds new GitHub Actions workflows to enable Continuous Integration using SBT based on #1764 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

File: project/CelebornBuild.scala [hunk 1]
```diff
@@ -553,7 +573,7 @@ trait SparkClientProjects {
         ) ++ commonUnitTestDependencies
       )
   }
-
```
Comment (pan3793) [inline project/CelebornBuild.scala:556]: should it be covered by spotless?

File: .github/workflows/sbt.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+#
+
+
+name: Celeborn SBT CI
+
+on:
+  push:
+    branches:
+      - main
+      - branch-*
```
Comment (pan3793) [inline .github/workflows/sbt.yml:25]: Emm ... I'm afraid we should exclude branch-0.3 and lower

Comment (cfmcgrady) [reply .github/workflows/sbt.yml:25]: Since we don't backport this to branch-0.3, I think it's safe.

File: project/CelebornBuild.scala (hunk 1 above)
Comment (cfmcgrady) [reply project/CelebornBuild.scala:556]: after experimentation, it seems that the maven spotless plugin is unable to format source files located outside the source directory.

Comment (pan3793) [reply project/CelebornBuild.scala:556]: not a big deal.

**Review by pan3793 — APPROVED** [review]

**Pull request closed without merging**